client/src/app/users/user-profile.component.ts

```diff
@@ -13,6 +13,7 @@ import { map, switchMap, takeUntil } from 'rxjs/operators';
 })
 export class UserProfileComponent implements OnInit, OnDestroy {
   user: User;
+  error: { help: string, httpResponse: string, message: string };
```

Comment:
We added this new `error` field to `UserProfileComponent` that holds the (currently three) parts of the error message. Those are then used in the HTML to display the different parts of the message.

There is an argument for having this be a proper, named class instead of an anonymous object like this, so feel free to push back on this.

Reply:
I don't have an immediate, gut reaction about using a proper, named class or not. Might that make the other error handling stuff easier?

Reply:
Maybe? It depends on whether this sort of structure is re-usable in other settings.

Maybe that's the way to deal with this – leave it alone for now, and see if we end up creating the same or something similar elsewhere. If we do, extract that out into a named class. If we don't, then leave it alone.